### PR TITLE
fix a bug that proguard configuration keeps all class names.

### LINKED ISSUE #3689: [WARNING!] Obfuscation is not working with 2.1.0
**Beware, don't release to Google Play with Realm 2.1.0** if you want your code to be obfuscated. Realm 2.1.0 won't allow any of your class names to be obfuscated, even on classes that are not RealmObjects.

I'm using custom dictionaries for it, but it probably does not matter. I tested this multiple times and my code with 2.0.2 is always correctly obfuscated, code with 2.1.0 is not obfuscated at all. Luckily this happened to me once with another library and I'm checking the resulting apk with an apktool before releasing.

### PR DESCRIPTION
**Pull request by zaki50**
Our ProGuard configuration introduced in `2.1.0` unexpectedly keeps classes those do not have any fields with KeepMember annotation.

fixes #3689

### CODE REVIEW

**Comment by zaki50** [comment]
review this please @realm/java

File: realm/realm-library/src/main/java/io/realm/log/RealmLogger.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
  * Interface for custom loggers that can be registered at {@link RealmLog#add(RealmLogger)}.
  * The different log levels are described in {@link LogLevel}.
  */
+@KeepMember
```
Comment (beeender) [inline realm/realm-library/src/main/java/io/realm/log/RealmLogger.java]: So the `@KeepMember` has to be applied to the class as well? Is it still necessary to apply `@KeepMember` for the `log` method below?

Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/log/RealmLogger.java]: yes.
I'd like to achieve only requiring to add the annotation to methods, but I couldn't find the way.

**Comment by beeender** [comment]
LGTM, but I know almost nothing about proguard :P

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 2.1.1
+
+### bug fixes
```
Comment (cmelchior) [inline CHANGELOG.md]: `bug fixes` -> `Bug Fixes`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
...
+
+* ProGuard configuration introduced in 2.1.0 unexpectedly keeps classes those do not have any fields with KeepMember annotation (#3689).
```
Comment (cmelchior) [inline CHANGELOG.md]: kept classes that did not have the @KeepMember annotation

File: realm/realm-library/src/main/java/io/realm/internal/KeepMember.java [hunk 4]
```diff
@@ -24,9 +24,9 @@
 /**
  * This annotation is used to mark the fields and methods to be kept by ProGuard/DexGuard.
  * The ProGuard configuration must have '-keep class io.realm.internal.KeepMember'
- * and '-keep,includedescriptorclasses class * { @io.realm.internal.KeepMember *; }'.
+ * and '-keep,includedescriptorclasses @io.realm.internal.KeepMember class * { @io.realm.internal.KeepMember *; }'.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/internal/KeepMember.java]: Also add that this annotation must be added to class as well to work

**Comment by zaki50** [comment]
All comments are addressed

**Pull request merged**